### PR TITLE
WIP: Addition of credential creation

### LINKED ISSUE #93: Generate credentials
As an awx-resource user I would like to generate credentials so that I can use them in my jobs and plays.

### PR DESCRIPTION
**Pull request by cooktheryan**
closes #93

### CODE REVIEW

**Comment by cooktheryan** [comment]
image

File: PROJECT [hunk 1]
```diff
@@ -34,4 +34,11 @@ resources:
   group: tower
   kind: AnsibleWorkflow
   version: v1alpha1
+- api:
+    crdVersion: v1
+    namespaced: true
+  domain: ansible.com
+  group: tower
+  kind: AnsibleCredentials
```
Comment (rooftopcellist) [inline PROJECT:42]: Can we make this singular everywhere for consistency?

File: config/crd/bases/tower.ansible.com_ansiblecredentials.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: ansiblecredentials.tower.ansible.com
```
Comment (cooktheryan) [inline config/crd/bases/tower.ansible.com_ansiblecredentials.yaml:5]: credential not credentials

File: config/crd/bases/tower.ansible.com_ansibleprojects.yaml [hunk 3]
```diff
@@ -30,7 +30,30 @@ spec:
...
+              project_organization:
+                type: string
+                description: The organization of the project
+              tower_auth_secret:
+                description: |
+                  A k8s secret that contains an access token for AWX. To create an access token see these docs: https://docs.ansible.com/automation-controller/4.1.0/html/userguide/applications_auth.html#add-tokens.
+                type: string
+            required:
+            - tower_auth_secret
+            - project_branch
```
Comment (cooktheryan) [inline config/crd/bases/tower.ansible.com_ansibleprojects.yaml]: See if I can make these options non-git friendly

File: roles/credentials/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+Role Name
+=========
+
+A brief description of the role goes here.
+
+Requirements
+------------
+
+Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance,
```
Comment (rooftopcellist) [inline roles/credentials/README.md]: Can we remove some of this boilerplate and add in basic credentials docs?

**Comment by rooftopcellist** [comment]
@cooktheryan Looks like we need a couple changes:

- [ ] add `runner_image` & `runner_version`
- [ ] rename from `ansiblecredentials` --> `ansiblecredential`
- [ ] add in runner image pull policy var
- [ ] write test cases
- [ ] (optional) update boilerplate for roles

**Pull request closed without merging**